core/src/main/java/utilcalc/core/parser/Parser.java

```diff
@@ -0,0 +1,62 @@
+package utilcalc.core.parser;
+
+import static utilcalc.core.parser.ParserUtils.*;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.tomlj.*;
+import utilcalc.core.model.input.ReportInputs;
+
+public final class Parser {
+
+    private static final class GeneralSection {
+        private static final String SECTION_NAME = "general";
+        private static final String START_DATE = "start_date";
+        private static final String END_DATE = "end_date";
+        private static final String TENANT = "tenant";
+        private static final String OWNER = "owner";
+        private static final String REPORT_PLACE = "report_place";
+        private static final String REPORT_DATE = "report_date";
+        private static final String SOURCES = "sources";
+
+        private static final List<String> KNOWN_FIELDS =
+                List.of(START_DATE, END_DATE, TENANT, OWNER, REPORT_PLACE, REPORT_DATE, SOURCES);
+    }
+
+    private Parser() {}
+
+    public static ReportInputs parse(String input) {
+        TomlParseResult parseResult = Toml.parse(input);
+        if (parseResult.hasErrors()) {
+            String errorMessages =
+                    parseResult.errors().stream()
+                            .map(TomlParseError::toString)
+                            .collect(Collectors.joining("; "));
+            throw new ParsingException("Syntax error: " + errorMessages);
+        }
+
+        try {
+            TomlTable general = requireNonNullTable(parseResult, GeneralSection.SECTION_NAME);
+
+            checkThatSectionContainsOnlyKnownFields(
+                    general, GeneralSection.KNOWN_FIELDS, GeneralSection.SECTION_NAME);
+
+            LocalDate startDate = requireNonNullLocalDate(general, GeneralSection.START_DATE);
+            LocalDate endDate = requireNonNullLocalDate(general, GeneralSection.END_DATE);
+            List<String> tenant = requireNonNullList(general, GeneralSection.TENANT, String.class);
+            List<String> owner = requireNonNullList(general, GeneralSection.OWNER, String.class);
+            String reportPlace = requireNonNullString(general, GeneralSection.REPORT_PLACE);
+            LocalDate reportDate = requireNonNullLocalDate(general, GeneralSection.REPORT_DATE);
+            List<String> sources =
```

Comment:
Neni to sice v analyze explicitne, ale `sources` by melo byt optional (je tam jen zminka o tom, ze muze byt prazdna, pak ale neni duvod, aby to uzivatel do souboru vubec daval).